### PR TITLE
Add desktop and icon rules to Makefile, document icon installation

### PR DESCRIPTION
**Pull request by GRFreire**
Is there a reason why the `.desktop` file and the icon isn't installed when running `make install`? Well, I assume there is not, thus this PR.

### CODE REVIEW

**Comment by N-R-K** [comment]
https://github.com/nsxiv/nsxiv/commit/56b6d23c0cb05973415bdd2c66af26e8f7db2539
Seems like Muennich was against installing desktop file by default. I am assuming this maybe due to there being user's who don't want the .desktop file.

As for icons, I think it might be more appropriate to add instruction on how to install them on the README.

**Comment by N-R-K** [comment]
How about this

* `make install` - stays same
* `make desktop` - installs the .desktop file
* `make uninstall` - stays same, but also removes .desktop

And for the icons, we just the command to the README.

**Comment by GRFreire** [comment]
I find it easier to install by default, but I see your point (and it makes sense). Lets see if anyone will say something otherwise I'll change the `makefile` accordingly.

File: Makefile [hunk 1]
```diff
@@ -85,6 +85,10 @@ install: all
 	sed "s!PREFIX!$(PREFIX)!g; s!VERSION!$(VERSION)!g" nsxiv.1 \
 		>$(DESTDIR)$(MANPREFIX)/man1/nsxiv.1
 	chmod 644 $(DESTDIR)$(MANPREFIX)/man1/nsxiv.1
+	@echo "INSTALL nsxiv.desktop"
+	install -Dt $(DESTDIR)$(PREFIX)/share/applications nsxiv.desktop
```
Comment (N-R-K) [inline Makefile:89]: We're not using `install` anymore, as it's not POSIX. Try `mkdir -p` and then `cp` instead.

**Comment by N-R-K** [comment]
> I find it easier to install by default

I don't disagree, but it's easier to run `make desktop` for someone that wants it, than it is to run `rm /path/to/desktop/nsxiv.desktop` for someone that doesn't. Also he'll need to `rm` it after every update.

**Comment by GRFreire** [comment]
> How about this
> 
>     * `make install` - stays same
> 
>     * `make desktop` - installs the .desktop file
> 
>     * `make uninstall` - stays same, but also removes .desktop
> 
> 
> And for the icons, we just add the command to the README.

I was thinking here, usually all make commands you can run without root privileges and it will not affect your system, except for `make install` and `make uninstall`. Having `make desktop` gives the impression that it will only "builld" the desktop file, not install it. Also, how would you uninstall the desktop file? Is there a way of having `make desktop (un)install` a command inside make?

**Comment by N-R-K** [comment]
> how would you uninstall the desktop file?

`make uninstall` should clean up the .desktop files too.

- - -

On a sidenote, I realized that the .desktop and icon files aren't really going to change much if at all. I think that might've been why Muennich didn't install/uninstall them by default.

**Comment by N-R-K** [comment]
If one more maintainer thinks that having .desktop install by default is preferable, then I won't object it.
I'm just trying to think of some of the possible reasons on why Muennich didn't install it by default.

**Comment by GRFreire** [comment]
I realized that .desktop was missing when I was trying to test something and nsxiv wasn't showing up on my graphical file manager as a program to open the image

**Comment by Ytrewq13** [comment]
I think I agree with @N-R-K on this. Installing the .desktop file by default would cause too much annoyance for those users who don't want it.

Maybe `make desktop` would install only the .desktop file, and we add an environment variable that indicates if the user wants to install the .desktop file with their `make install` command? Something like: `make install desktop=yes`

**Comment by N-R-K** [comment]
Nah, env var for someone as simple as this is too much. Just `make desktop` is fine. Remember that .desktop is someone you'll need to install only once, it's highly unlikely for it to change. Same with icons.

File: README.md [hunk 2]
```diff
@@ -76,9 +76,13 @@ Installing nsxiv:
 
     # make install
 
-Please note, that this requires root privileges.
+Installing icons:
+
+	# make -C icon install
```
Comment (N-R-K) [inline README.md:81]: We're currently using 4 spaces for codeblocks in README.
Maybe we should switch to tabs... Anyways that should be it's own PR. For now let's keep things unified.
```suggestion
    # make -C icon install
```

File: README.md [hunk 3]
```diff
@@ -76,9 +76,13 @@ Installing nsxiv:
...
+
+Please note, that these requires root privileges.
 By default, nsxiv is installed using the prefix "/usr/local", so the full path
-of the executable will be "/usr/local/bin/nsxiv".
+of the executable will be "/usr/local/bin/nsxiv" and the icon path will be "/usr/local/share/icons/hicolor/{size}/apps/nsxiv.png".
```
Comment (N-R-K) [inline README.md:85]: I think we're on 80 char line limit.
```suggestion
of the executable will be "/usr/local/bin/nsxiv" and the icon path will be
"/usr/local/share/icons/hicolor/{size}/apps/nsxiv.png".
```

**Review by N-R-K — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -76,9 +76,18 @@ Installing nsxiv:
...
+
+Installing icons:
+
+    # make -C icon install
+
+Please note, that these requires root privileges.
 By default, nsxiv is installed using the prefix "/usr/local", so the full path
-of the executable will be "/usr/local/bin/nsxiv".
+of the executable will be "/usr/local/bin/nsxiv" and the icon path will be
+"/usr/local/share/icons/hicolor/{size}/apps/nsxiv.png".
```
Comment (N-R-K) [inline README.md:90]:
```suggestion
of the executable will be "/usr/local/bin/nsxiv", the `.desktop` entry will be
"/usr/local/share/applications/nsxiv.desktop" and the icon path will be
"/usr/local/share/icons/hicolor/{size}/apps/nsxiv.png".
```

**Review by N-R-K — CHANGES_REQUESTED** [review]
If we're mentioning the icon path, might as well mention the `.desktop` path as well.
Otherwise LGTM.

**Review by N-R-K — APPROVED** [review]

**Comment by XPhyro** [comment]
```
make -C icon install
```

What's the reason we don't include an `icon` target in the root Makefile?

**Comment by N-R-K** [comment]
> What's the reason we don't include an `icon` target in the root Makefile?

Good call. I guess we can have a `make icon` and give it the same treatment as `.desktop` when uninstalling.

**Comment by N-R-K** [comment]
@GRFreire Can you allow "edits by maintainer"? I have a patch that adds `make icon` to root Makefile and removes `icon/Makefile` but I can't push.

**Comment by N-R-K** [comment]
@XPhyro pushed https://github.com/nsxiv/nsxiv/pull/80/commits/9d9ea34c679f39ad58d9b0de4f2291fa38fe8f9a , we should be good now.

**Review by XPhyro — APPROVED** [review]

**Comment by TAAPArthur** [comment]
@GRFreire @N-R-K @XPhyro 

Maybe it is just me but I'd expect `make x` to not install anything to / unless x starts with install. I would think `make desktop` and `make icon` would just generate the desktop file/icons respectively.
Thoughts on `make install-desktop` and `make install-icon`?

**Comment by GRFreire** [comment]
> Maybe it is just me but I'd expect `make x` to not install anything to / unless x starts with install. I would think `make desktop` and `make icon` would just generate the desktop file/icons respectively.

I was wondering the same thing, but thought it was just a me thing as well

> @GRFreire: I was thinking here, usually all make commands you can run without root privileges and it will not affect your system, except for `make install` and `make uninstall`

---

> Thoughts on `make install-desktop` and `make install-icon`?

I think that this is a good suggestion, but I dont mind leaving the way it was already merged. Let's see what the others have to say.

**Comment by N-R-K** [comment]
I had thought about that too but didn't suggest it since it's a bit longer to type. But since you won't be installing desktop and icons too often, I think my concern was invalid.

So yeah, having `install-` prefix is okay by me.

**Comment by XPhyro** [comment]
I suggest also having an `install-all` that does `install install-icon install-desktop`.

**Comment by XPhyro** [comment]
See #96.

**Pull request merged**